internal/monitors/replicationstatus.go

```diff
@@ -0,0 +1,61 @@
+package monitors
+
+import (
+	"log"
+	"time"
+
+	"cloudant.com/cloudant_exporter/internal/utils"
+	"github.com/IBM/cloudant-go-sdk/cloudantv1"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+type ReplicationStatusMonitor struct {
+	Cldt     *cloudantv1.CloudantV1
+	Interval time.Duration
+	FailBox  *utils.FailBox
+}
+
+var (
+	replicatonStatus = promauto.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "cloudant_replication_status",
+			Help: "Current status per replication",
+		},
+		[]string{"docid", "source", "target", "status"},
+	)
+)
+
+func (rc *ReplicationStatusMonitor) Name() string {
+	return "ReplicationStatusMonitor"
+}
+
+func (rc *ReplicationStatusMonitor) Retrieve() error {
+	var skip int = 0
+	var batchSize int = 100
+
+	// repeat until we get a smaller batch than we asked for
+	for {
+		// fetch scheduler jobs
+		getSchedulerDocsOptions := rc.Cldt.NewGetSchedulerDocsOptions()
+		getSchedulerDocsOptions.SetLimit(int64(batchSize))
+		getSchedulerDocsOptions.SetSkip(int64(skip))
+
+		schedulerJobsResult, _, err := rc.Cldt.GetSchedulerDocs(getSchedulerDocsOptions)
+		if err != nil {
+			return err
+		}
+		for _, d := range schedulerJobsResult.Docs {
+			log.Printf("[ReplicationStatusMonitor] docId %q source %q target %q status %q", *d.DocID, *d.Source, *d.Target, *d.State)
+			replicatonStatus.WithLabelValues(*d.DocID, *d.Source, *d.Target, *d.State).Set(0)
```

Comment:
You'll want this to be `Set(1)` I think, to support summing the replications by whatever label in the metrics. Otherwise any summing will give you zero :)